src/main/java/org/opensearch/ubi/UbiActionFilter.java

```diff
@@ -104,82 +102,100 @@ public <Request extends ActionRequest, Response extends ActionResponse> void app
+                listener.onFailure(ex);
+            }
 
-                        for (final SearchHit hit : ((SearchResponse) response).getHits()) {
+        });
 
-                            if (objectIdField == null || objectIdField.isEmpty()) {
-                                // Use the result's docId since no object_id was given for the search.
-                                queryResponseHitIds.add(String.valueOf(hit.docId()));
-                            } else {
-                                final Map<String, Object> source = hit.getSourceAsMap();
-                                queryResponseHitIds.add((String) source.get(objectIdField));
-                            }
+    }
 
-                        }
+    private ActionResponse handleSearchRequest(final SearchRequest searchRequest, ActionResponse response) {
 
-                        final String queryResponseId = UUID.randomUUID().toString();
-                        final QueryResponse queryResponse = new QueryResponse(queryId, queryResponseId, queryResponseHitIds);
-                        final QueryRequest queryRequest = new QueryRequest(queryId, userQuery, userId, query, queryAttributes, queryResponse);
+        if (response instanceof SearchResponse) {
 
-                        final String dataPrepperUrl = environment.settings().get(UbiSettings.DATA_PREPPER_URL);
-                        if(dataPrepperUrl != null) {
-                            sendToDataPrepper(dataPrepperUrl, queryRequest);
-                        } else {
-                            indexQuery(queryRequest);
-                        }
+            final UbiParameters ubiParameters = UbiParameters.getUbiParameters(searchRequest);
 
-                        SearchResponse searchResponse = (SearchResponse) response;
+            if (ubiParameters != null) {
 
-                        response = (Response) new UbiSearchResponse(
-                                searchResponse.getInternalResponse(),
-                                searchResponse.getScrollId(),
-                                searchResponse.getTotalShards(),
-                                searchResponse.getSuccessfulShards(),
-                                searchResponse.getSkippedShards(),
-                                searchResponse.getTook().millis(),
-                                searchResponse.getShardFailures(),
-                                searchResponse.getClusters(),
-                                queryId
-                        );
+                final String queryId = ubiParameters.getQueryId();
+                final String userQuery = ubiParameters.getUserQuery();
+                final String userId = ubiParameters.getClientId();
+                final String objectIdField = ubiParameters.getObjectIdField();
+                final Map<String, String> queryAttributes = ubiParameters.getQueryAttributes();
 
+                // TODO: Ignore the UBI in ext.
```

Comment:
What does this mean?   I'm fine with the TODO, but would love a bit more detail about what this means ;-)

Reply:
The thought was to not log the `ubi` part of the query to save space. I will take this out and write an issue.